### PR TITLE
Add debug button to identify unread conversations

### PR DESCRIPTION
**Pull request by marcoconti83**
In some situation it’s not possible to find which conversations are causing the badge count to be non-zero, even if they all look read. This option in the debug menu will allow to identify which one is the offending conversation. Hopefully this will allow us to understand the pattern of this issue.

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/SettingsCellDescriptorFactory.swift [hunk 1]
```diff
@@ -310,4 +295,39 @@ import Foundation
...
     }
     
+    private static func findUnreadConversation(_ type: SettingsCellDescriptorType) {
+        guard let userSession = ZMUserSession.shared() else { return }
+        let predicate = ZMConversation.predicateForConversationConsideredUnread()!
+        
+        let message: String
+        if let convo = (ZMConversationList.conversations(inUserSession: userSession) as! [ZMConversation])
+            .first(where: { predicate.evaluate(with: $0) }) {
+            message = ["Found an unread conversation:",
```
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/SettingsCellDescriptorFactory.swift]: Good start, may be you can add the button to copy the conversation description in the alert view?

File: Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/SettingsCellDescriptorFactory.swift [hunk 2]
```diff
@@ -310,4 +295,39 @@ import Foundation
...
+                       "\(convo.displayName)",
+                        "<\(convo.remoteIdentifier?.uuidString ?? "n/a")>"
+            ].joined(separator: "\n")
+        } else {
+            message = "No unread conversation"
+        }
+        UIAlertView(title: nil, message: message, delegate: nil, cancelButtonTitle: "OK").show()
```
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/SettingsCellDescriptorFactory.swift]: We are moving from UIAlertView (it produces the deprecation warning). Please use `UIAlertController`.

Comment (marcoconti83) [inline Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/SettingsCellDescriptorFactory.swift]: Using `UIAlertController` requires having a reference to a VC to call `UIViewController.presentViewController(alert)`. Is there a shared instance I should use, or do I need to inject one into this `SettingsCellDescriptorFactory`?

File: Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/SettingsCellDescriptorFactory.swift (hunk 1 above)
Comment (marcoconti83) [inline Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/SettingsCellDescriptorFactory.swift]: What do you mean by "to copy the conversation description"? You mean copy to the clipboard, to be pasted somewhere else?

File: Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/SettingsCellDescriptorFactory.swift (hunk 2 above)
Comment (marcoconti83) [inline Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/SettingsCellDescriptorFactory.swift]: Found this:
```
        guard let controller = UIApplication.shared.wr_topmostController(onlyFullScreen: false) else { return }
```

**Comment by mikeger** [comment]
Yes, this should work! O was thinking to be able to copy the description so it can be pasted to the text editor.

Sent from a Mobile Device

**Comment by marcoconti83** [comment]
Updated

**Pull request merged**